docs/architecture/psa-migration/strategy.md

```diff
@@ -355,6 +361,31 @@ the new macros in the hope of avoiding confusion.
 Executing step 3 will mostly consist of using the right dependency macros in
 the right places (once the previous steps are done).
 
+**Note on testing**
+
+Since supporting driver-only builds is not about adding features, but also
+supporting existing features in new types of builds, the existing tests should
+be enough to cover everything, as long as they're run in the newly supported
+configurations. An all.sh component needs to be present each main type of
+configutation we support (for example, with this (family of) algorithm
```

Comment:
This sentence is a bit hard to understand. Is an `all.sh` component an entity that we use anywhere outside than in `all.sh`?  What does it mean that it has to be present in each main types of configurations? 
Example: `component_test_full_cmake_gcc_asan` - does this (in reference to this sentence) mean that it should work with both the drivers and software implementations?

Reply:
Ah, sorry, that was unclear indeed. I'm not trying to say that each `all.sh` component should work with drivers, I'm trying to say that when we add support for X provided by drivers, we need to test that somewhere in `all.sh`. So, basically, that this PR had to add `component_test_psa_crypto_config_accel_hash_use_psa` - the goal of the PR is to make some things work with driver-only hashes and `USE_PSA`, so it has to test a build with driver-only hashes and `USE_PSA`.

So really this paragraph is not saying more than "Things should be tested. New configurations are not tested by adding test cases in the test suites, but by adding components in `all.sh`."